packages/data-broker/src/DataBroker.js

```diff
@@ -303,4 +298,71 @@ export class DataBroker {
       dataServiceMapping,
     };
   }
+
+  async getPulls(dataSources, orgUnitCodes) {
+    const orgUnits = await this.models.entity.find({ code: orgUnitCodes });
+
+    // Note: each service will pull for ALL org units and ALL data sources.
```

Comment:
Is there a reason we can't fetch for just the countries/data elements in each mapping? We work out which countries and dataElements/groups belong to each along the way, so we'd just need to combine them when building the `uniqueMapping`?

It feels a little odd to pull for dataElements and countries that we **know** aren't in the data service (and also could create issues if they actually are in the data service but Tupaia isn't configured to fetch them from there...)

Reply:
Discussed.